Kernel/Locking/Mutex.cpp

```diff
@@ -12,15 +12,19 @@
 #include <Kernel/Locking/Spinlock.h>
 #include <Kernel/Thread.h>
 
+extern bool g_in_early_boot;
+
 namespace Kernel {
 
 void Mutex::lock(Mode mode, [[maybe_unused]] LockLocation const& location)
 {
     // NOTE: This may be called from an interrupt handler (not an IRQ handler)
     // and also from within critical sections!
     VERIFY(!Processor::current_in_irq());
-    if constexpr (LOCK_IN_CRITICAL_DEBUG)
-        VERIFY_INTERRUPTS_ENABLED();
+    // There are no interrupts enabled in early boot.
+    if (!g_in_early_boot)
+        if constexpr (LOCK_IN_CRITICAL_DEBUG)
+            VERIFY_INTERRUPTS_ENABLED();
```

Comment:
1. Please add braces around an `if` block with more than one line
2. I'd switch the `constexpr` and `g_in_early_boot` lines: that way it's clear that both lines are guarded by the debug flag (which eventually needs to be removed)